src/App.tsx

```diff
@@ -14,11 +15,100 @@ const goodsFromServer: string[] = [
   'Garlic',
 ];
 
-const App: React.FC = () => (
-  <div className="App">
-    <h1>Selected good: -</h1>
-    {goodsFromServer.length}
-  </div>
-);
+type State = {
+  selectedGood: string[],
+};
 
-export default App;
+type Props = {};
+
+export class App extends React.Component<Props, State> {
+  state = {
+    selectedGood: ['Jam'],
+  };
+
+  arr = this.state.selectedGood;
+
+  addGood = (good: string) => {
+    this.setState((state) => ({
+      selectedGood: [...state.selectedGood, good],
+    }));
+  };
+
+  removeGood = (good: string) => {
+    this.setState((state: State) => {
+      state.selectedGood.splice(state.selectedGood.indexOf(good), 1);
+
+      return { selectedGood: state.selectedGood };
+    });
```

Comment:
splice modified existing arr. Can you instead of using setState, modify this.state.selectedGood?